controller/src/beerocks/master/db/db.cpp

```diff
@@ -1004,6 +1004,36 @@ std::set<std::string> db::get_node_children(std::string mac, int type, int state
     return children_macs;
 }
 
+std::list<sMacAddr> db::get_1905_1_neighbors(const sMacAddr &al_mac)
+{
+    auto al_mac_str = network_utils::mac_to_string(al_mac);
+    std::list<sMacAddr> neighbors_al_macs;
+    auto all_al_macs = get_nodes(beerocks::TYPE_IRE);
+
+    // According to IEEE 1905.1 a neighbor is defined as a first circle only, so we need to filter
+    // out the childrens from second circle and above.
+    for (const auto &al_mac_iter : all_al_macs) {
+        if (get_node_parent_ire(al_mac_iter) != al_mac_str) {
+            continue;
+        }
+        neighbors_al_macs.push_back(network_utils::mac_from_string(al_mac_iter));
```

Comment:
It would feel more natural to me to invert the condition:
```suggestion
        if (get_node_parent_ire(al_mac_iter) == al_mac_str) {
            neighbors_al_macs.push_back(network_utils::mac_from_string(al_mac_iter));
        }
```

Reply:
ok